roles/openshift_health_checker/openshift_checks/disk_availability.py

```diff
@@ -0,0 +1,121 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, get_var
+
+
+class DiskAvailability(OpenShiftCheck):
+    """Check that recommended disk space is available."""
+
+    name = "disk_availability"
+    tags = ["preflight"]
+
+    recommended_diskspace = {
+        "nodes": 15*10**9,
+        "masters": 40*10**9,
+        "etcd": 20*10**9,
+    }
+
+    minimum_docker_space_percent = 5.0
+    supported_docker_dev_driver = "devicemapper"
+
+    def run(self, tmp, task_vars):
+        self.minimum_docker_space_percent = get_var(task_vars,
+                                                    "healthcheck_minimum_docker_diskspace_percent",
+                                                    default=self.minimum_docker_space_percent)
+
+        is_containerized = get_var(task_vars, "openshift", "common", "is_containerized")
+        ansible_mounts = get_var(task_vars, "ansible_mounts")
+
+        if is_containerized:
+            failed, msg = self.containerized_volume_check(task_vars)
+            if failed:
+                return {"failed": True, "msg": msg}
+
+            return {"changed": False}
+
+        failed, msg = self.noncontainerized_volume_check(ansible_mounts, task_vars)
+        return {"failed": failed, "msg": msg}
+
+    def containerized_volume_check(self, task_vars):
+        docker_facts = self.module_executor("docker_facts", {}, task_vars)
+        info = docker_facts.get("docker_facts", {})
+        status = dict(info.get("DriverStatus", {}))
+
+        if not status:
+            msg = "Unable to retrieve storage driver information for {driver}."
+            return {"failed": True, "msg": msg.format(driver=self.supported_docker_dev_driver)}
+
+        # ensure that the docker host is using a storage driver that our installation
+        # is known to support. The only supported driver is currently "devicemapper"
+        if info.get("Driver", "") == self.supported_docker_dev_driver:
+            disktotal = self.to_bytes(status.get("Data Space Total", ""))
+            diskavail = self.to_bytes(status.get("Data Space Available", ""))
+
+            if not disktotal or not diskavail:
+                msg = "Unable to retrieve storage driver information for {driver}."
+                return True, msg.format(driver=self.supported_docker_dev_driver)
+
+            diskleft_percent = diskavail / disktotal * 100
```

Comment:
Ah, I was not sure if it would be approproate to use the [master/node](https://docs.openshift.org/latest/install_config/install/prerequisites.html#system-requirements) minimum diskpace requirements for the containerized check as well, therefore I resorted to a percentage (check that docker volume storage is not at > 90% full for example)